src/main/java/com/robertx22/age_of_exile/capability/player/PlayerData.java

```diff
@@ -172,7 +176,16 @@ public void deserializeNBT(CompoundTag nbt) {
 
         this.team = loadOrBlank(TeamData.class, new TeamData(), nbt, TEAM_DATA, new TeamData());
         this.prophecy = loadOrBlank(PlayerProphecies.class, new PlayerProphecies(), nbt, PROPHECY, new PlayerProphecies());
-        this.talents = loadOrBlank(TalentsData.class, new TalentsData(), nbt, TALENTS_DATA, new TalentsData());
+        TalentsData newTalentsData = loadOrBlank(TalentsData.class, new TalentsData(), nbt, TALENTS_DATA, new TalentsData());
+        var newData = newTalentsData.getPerks().values().stream().map(SchoolData::getAllocatedPoints).collect(Collectors.toSet());
+        var oldData = this.talents.getPerks().values().stream().map(SchoolData::getAllocatedPoints).collect(Collectors.toSet());
```

Comment:
I don't like this. It's here to check if a talent was allocated? Why not just set dirty whenever a talent is added/removed?

Reply:
I think I can do the same trick as the button here, just let the connection renderer wait for the update. 
maybe dirty sync is also a solution, I'll try.